docs/articles/documentation/guides/basic-guides/select-page-elements.md

```diff
@@ -0,0 +1,542 @@
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('DOM Node Snapshot', async t => {
+    const sliderHandle         = Selector('#slider').child('span');
+    const sliderHandleSnapshot = await sliderHandle();
+
+    console.log(sliderHandleSnapshot.hasClass('ui-slider-handle'));    // => true
+    console.log(sliderHandleSnapshot.childElementCount);               // => 0
+});
+```
+
+Note that if a selector matches several DOM nodes on the page, this asynchronous call returns the first node from the matched set.
+
+> TestCafe wait mechanisms do not work when you pass a snapshot's property to an [assertion](assert.md). To enable [Smart Assertion Query]assert.md#smart-assertion-query-mechanism), pass [selector's properties](../../reference/test-api/domnodestate.md#members-common-across-all-nodes) instead.
+
+### Define Action Targets
+
+You can pass selectors to [test actions](interact-with-the-page.md) to specify the action's target element.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+const label = Selector('#tried-section').child('label');
+
+test('My Test', async t => {
+    await t.click(label);
+});
+```
+
+If the selector matches multiple DOM nodes, the action is executed for the first node.
+
+Before an action is performed, TestCafe waits for the target element to appear in the DOM and become visible. If this does not happen within the [selector timeout](#selector-timeout), the test fails.
+
+You can also pass DOM element snapshots to test actions.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+const label = Selector('#tried-section').child('label');
+
+test('My Test', async t => {
+    const labelSnapshot = await label();
+
+    await t.click(labelSnapshot);
+});
+```
+
+In this instance, the selector that fetched this snapshot is re-executed before the action.
+
+### Define Assertion Actual Value
+
+You can pass a selector property to an [assertion](assert.md) to check whether the DOM node has the expected state.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('Assertion with Selector', async t => {
+    const developerNameInput = Selector('#developer-name');
+
+    await t
+        .expect(developerNameInput.value).eql('')
+        .typeText(developerNameInput, 'Peter')
+        .expect(developerNameInput.value).eql('Peter');
+});
+```
+
+When you pass selector properties instead of values, TestCafe enables [Smart Assertion Query Mechanism](assert.md#smart-assertion-query-mechanism) to avoid accidental errors and unstable tests. In this example, the assertion does not fail immediately if the condition does not match. TestCafe recalculates the `developerNameInput.value` property value until the assertion passes or the timeout expires.
```

Comment:
How can you avoid 'accidental errors'?

Reply:
okay, so we can *avoid random errors* I guess